### PR TITLE
Broadcast header on subroundBlock, on the shard_meta topic

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- header can be broadcast during subround block with the equivalent proofs
  
## Proposed changes
- implement needed changes

## Testing procedure
- will be tested with feat branch

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: consensus/spos/bls/subroundBlock.go [hunk 1]
```diff
@@ -318,22 +323,56 @@ func (sr *subroundBlock) sendBlockBody(
 // sendBlockHeader method sends the proposed block header in the subround Block
 func (sr *subroundBlock) sendBlockHeader(
 	headerHandler data.HeaderHandler,
-	marshalizedHeader []byte,
+	marshalledHeader []byte,
 	signature []byte,
 ) bool {
-	headerHash := sr.Hasher().Compute(string(marshalizedHeader))
+	if !sr.EnableEpochsHandler().IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, headerHandler.GetEpoch()) {
```
Comment (ssd04) [inline consensus/spos/bls/subroundBlock.go:329]: add `TODO[cleanup cns finality]` comment here also?

File: consensus/spos/bls/subroundEndRound.go [hunk 2]
```diff
@@ -970,12 +972,16 @@ func (sr *subroundEndRound) updateMetricsForLeader() {
 }
 
 func (sr *subroundEndRound) broadcastBlockDataLeader(sender []byte) error {
+	// TODO[cleanup cns finality]: remove this method, block data was already broadcast during subroundBlock
```
Comment (ssd04) [inline consensus/spos/bls/subroundEndRound.go:975]: will this be done in a future PR? maybe set the comment only when the changed wil be implemented

File: consensus/spos/bls/subroundBlock.go (hunk 1 above)
Comment (sstanculeanu) [reply consensus/spos/bls/subroundBlock.go:329]: todo was added on sendBlockHeaderBeforeEquivalentProofs method, to be deleted

File: consensus/spos/bls/subroundEndRound.go (hunk 2 above)
Comment (sstanculeanu) [reply consensus/spos/bls/subroundEndRound.go:975]: it is already implemented.. body and header are sent on subround block. they won't be modified after

**Review by ssd04 — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**